client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -77,7 +77,7 @@
  *
  * Requests can be traced by enabling trace logging for "tracer". The trace logger outputs requests and responses in curl format.
  */
-public final class RestClient implements Closeable {
+public class RestClient implements Closeable {
```

Comment:
Its constructor is package private which makes this not a huge deal, but I wouldn't want people to think that they can subclass this class. I think this is more important than being able to mock it. I got around this by mocking the underlying http client instead and still building a proper RestClient through RestClientBuilder. Do you think that is too much to ask?